woocommerce/woocommerce-gutenberg-products-block
# Foundation of New Product Collection Block

This PR introduces the foundations for a new Product Collection block. It includes the creation of two new blocks:
1. Product Collection
2. Product Template

The Product Collection block will replace the existing Products (Beta) block. When the Product Collection block is added to the Editor, it will display all products in a grid view. On the frontend, all products should be displayed, and pagination should function as expected.



Fixes #9267


### Testing

1. Create a new post 
2. Add the Product Collection block to the post.
3. Verify that the UI of the Product Collection block matches that of the existing Products Beta block in both the editor and the frontend.
4. Test the pagination functionality on the frontend to ensure it works as expected.

Here is a quick demo:

https://user-images.githubusercontent.com/16707866/236784525-9bb93c6d-e538-40d3-9524-4d71f91d4431.mov


### WooCommerce Visibility




### Changelog

> Foundation of New `Product Collection` Block
